ts/features/itwallet/common/saga/index.ts

```diff
@@ -5,10 +5,21 @@ import { checkCredentialsStatusAttestation } from "../../credentials/saga/checkC
 import { handleWalletCredentialsRehydration } from "../../credentials/saga/handleWalletCredentialsRehydration";
 import { watchItwLifecycleSaga } from "../../lifecycle/saga";
 import { warmUpIntegrityServiceSaga } from "../../lifecycle/saga/checkIntegrityServiceReadySaga";
-import { checkWalletInstanceStateSaga } from "../../lifecycle/saga/checkWalletInstanceStateSaga";
+import {
+  checkWalletInstanceInconsistencySaga,
+  checkWalletInstanceStateSaga
+} from "../../lifecycle/saga/checkWalletInstanceStateSaga";
 import { checkCurrentWalletInstanceStateSaga } from "../../lifecycle/saga/checkCurrentWalletInstanceStateSaga.ts";
 
 export function* watchItwSaga(): SagaIterator {
+  const isWalletInstanceConsistent = yield* call(
+    checkWalletInstanceInconsistencySaga
+  );
+
+  // If the wallet instance is inconsistent, we cannot proceed further.
+  if (!isWalletInstanceConsistent) {
+    return;
+  }
   yield* fork(warmUpIntegrityServiceSaga);
   yield* fork(watchItwLifecycleSaga);
```

Comment:
We should not cut off `warmUpIntegrityServiceSaga` and `watchItwLifecycleSaga`
I'd suggest a new logic order:

```suggestion
  yield* fork(warmUpIntegrityServiceSaga);
  yield* fork(watchItwLifecycleSaga);
  
  const isWalletInstanceConsistent = yield* call(
    checkWalletInstanceInconsistencySaga
  );

  // If the wallet instance is inconsistent, we cannot proceed further.
  if (!isWalletInstanceConsistent) {
    return;
  }
```

In this way we start integrity service warmup as soon as possibile, then, before anything else, we make sure the WI state is consistent. What do you think?

Reply:
Yes you are right. Anyway, thank you for giving me more context. Addressed in [074ee55](https://github.com/pagopa/io-app/pull/6852/commits/074ee55444556e36de4d6f1de5609d61e7001146) the correction